Physical_Engine/Query/BottomCentreline.cs

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.Engine.Reflection;
+using BH.Engine.Spatial;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Spatial.ShapeProfiles;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Configuration;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+
+        [Description("Returns the bottom centreline of an IFramingElement.")]
+        [Input("element", "The IFramingElement to query the bottom centreline of.")]
+        [Output("curve", "The bottom centreline of the IFramingElement.")]
+
```

Comment:
Delete this blank line

Reply:
*Commenting your main comment*

At least in the Revit to BHoM workflow when pulling IFramingElements from Revit the location line is always the elements centreline regardless of how it is defined in Revit. A tapered element will have a tilted location line a.s.o.

Reply:
That makes sense, but in a BHoM workflow we usually wouldn't mind too much about Revit at this stage of the code (even if those are the workflows in mind). @IsakNaslundBh what do you think regarding a method for `Centreline` of an `IFramingElement`? I'm mostly looking at a reduction of duplicated code, but if there's no benefit then this is acceptable :smile:

Reply:
As it stands right now, and as mentioned by @annastigenberg , the Location property of the FramingElements is the centreline already. We have had discussion of adding in offsets, insertionPoints and similar additional properties to the FramingElements, but until that is done, there is no need for a CentreLine method.

Not sure was the initial comment was about though?